segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/UtilsWrapper.java

```diff
@@ -0,0 +1,287 @@
+                                                    offsetToRead.get(),
+                                                    Math.toIntExact(Math.min(bytesRemaining.get(), buffer.length)),
+                                                    buffer,
+                                                    0)
+                                            .thenComposeAsync(bytesRead -> {
+                                                bytesRemaining.addAndGet(-bytesRead);
+                                                offsetToRead.addAndGet(bytesRead);
+                                                try {
+                                                    outputStream.write(buffer, 0, bytesRead);
+                                                    return completedFuture(null);
+                                                } catch (Exception e) {
+                                                    return CompletableFuture.failedFuture(e);
+                                                }
+                                            }, chunkedSegmentStorage.getExecutor()),
+                                        chunkedSegmentStorage.getExecutor()),
+                                    chunkedSegmentStorage.getExecutor());
+
+                }, chunkedSegmentStorage.getExecutor());
+
+    }
+
+    /**
+     * Overwrites the given chunk on the storage with given data.
+     *
+     * @param chunkName Name of the chunk to overwrite.
+     * @param inputStream {@link InputStream} which contains data to write.
+     * @param length number of bytes to write.
+     * @return A CompletableFuture that, when completed, will indicate that the operation completed.
+     *         If the operation failed, it will be completed with the appropriate exception.
+     */
+    public CompletableFuture<Void> overwriteChunk(String chunkName, InputStream inputStream, int length) {
+        Preconditions.checkNotNull(chunkName, "chunkName");
+        Preconditions.checkNotNull(inputStream, "inputStream");
+        return chunkedSegmentStorage.getChunkStorage().openWrite(chunkName)
+                .thenComposeAsync(deleteHandle -> chunkedSegmentStorage.getChunkStorage().delete(deleteHandle), chunkedSegmentStorage.getExecutor())
+                .thenRunAsync(() -> chunkedSegmentStorage.getChunkStorage().createWithContent(chunkName, length, inputStream), chunkedSegmentStorage.getExecutor());
+
+    }
+
+    /**
+     * Returns the list of {@link ExtendedChunkInfo} which contain data about all chunks for the segment.
+     *
+     * @param streamSegmentName Name of the segment.
+     * @param checkStorage Whether to retrieve information from underlying {@link ChunkStorage}.
+     * @return A CompletableFuture that, when completed, will contain a list of {@link ExtendedChunkInfo} objects associated with the segment.
+     *  If the operation failed, it will be completed with the appropriate exception. Notable Exceptions
```

Comment:
This looks like something is missing, like the `@throws`  annotation and something after `Notable Exceptions`. Please, pay attention to the javadoc.

Reply:
CompletableFuture returned will be failed future instead of throwing exception.
There are no notable exceptions here.

Reply:
I'll fix the comment.